### PR TITLE
Add IsStuckCondition, TransformAvailableCondition, and RateController tests

### PR DESCRIPTION
**Pull request by naiveHobo**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #1810 |
| Primary OS tested on | Ubuntu 18.04 |
| Robotic platform tested on | Turtlebot3 gazebo simulation |

---

## Description of contribution in a few bullet points

* Added tests for `IsStuckCondition`, `TransformAvailableCondition`, and `RateController`

### CODE REVIEW

File: nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/is_stuck_condition.hpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__IS_STUCK_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__IS_STUCK_CONDITION_HPP_
+
+// Copyright (c) 2018 Intel Corporation
```
Comment (SteveMacenski) [inline nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/is_stuck_condition.hpp:18]: Combine the 2 licenses with intels' name before Samsung

Comment (naiveHobo) [reply nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/is_stuck_condition.hpp:18]: Ah, sorry missed this completely. The original `is_stuck_condition` had only intel's license, the samsung stuff got copied here by mistake. I'll remove the samsung license altogether.

**Comment by naiveHobo** [comment]
> I assume for the files themselves, the cpp/hpp are just translations with no functional modifications?

Yup, direct translations.

**Comment by SteveMacenski** [comment]
@naiveHobo merge once CI fully passes

**Comment by naiveHobo** [comment]
@SteveMacenski coverage went from -0.48% to +56% after removing the license haha

**Comment by SteveMacenski** [comment]
I think its just a rebase issue since master has the other tasks in it this one doesn't have in its history and/or that CI issue we talked about. I've generally been tracking the coverage with a +/- 1% variance. As long as its trending up, I haven't been paying attention to small negative numbers when there's clearly tests written

**Comment by SteveMacenski** [comment]
I think that hits us to 69%, which is great. Super close to being in the 70s and now 3 packages with > 90% test coverage (up from 1 two weeks ago)

**Pull request merged**